### ISSUE TITLE
Intellisense is trying to open file with duplicated path on compilation error

### ISSUE THREAD

**Issue by McDic** [microsoft/vscode-cpptools#4062]
**Type: LanguageService**

**Describe the bug**
- OS and Version: Windows 10
- VS Code Version: July 2019 1.37.0
- C/C++ Extension Version: 0.25.0-insiders
- Other extensions you installed (and if the issue persists after disabling them): Green theme 0.0.2, Python(Made by MS) 2019.8.29288 - I disabled all when I encountered bug.
- A clear and concise description of what the bug is: VSCode is opening file in wrong path(duplicated path. VScode is trying to open `A/A` when actual file path is `A`) when I click compilation errors on "Problems tab".

**To Reproduce**

1. Install G++ 8.2.0, VSCode and C/C++ extension.
2. Make any folder and open with VSCode.
3. This is tasks.json.
```
{
    // See https://go.microsoft.com/fwlink/?LinkId=733558 
    // for the documentation about the tasks.json format
    "version": "2.0.0",
    "echoCommand": true,
    "tasks": [
        {
            "type": "shell",
            "label": "BUILD",
            "command": "C:\\MinGW\\bin\\g++.exe",
            "args": [
                "-g",
                "${file}",
                "-o",
                "${fileDirname}\\${fileBasenameNoExtension}.exe"
            ],
            /*"options": {
                "cwd": "C:\\MinGW\\bin"
            },*/
            "problemMatcher": [
                "$gcc"
            ],
            "group": "build"
        }
    ]
}
```
4. This is c_cpp_properties.json.
```
{
    "configurations": [
        {
            "name": "CompetitiveProgramming",
            "includePath": [
                "${workspaceFolder}/**"
            ],
            "defines": [
                "_DEBUG",
                "UNICODE",
                "_UNICODE",
                "__McDic__"
            ],
            "compilerPath": "C:\\MinGW\\bin\\g++.exe",
            "intelliSenseMode": "gcc-x64",
            "cStandard": "c11",
            "cppStandard": "c++17"
        }
    ],
    "version": 4
}
```
5. This is my cpp code, which will make compilation error. (**Line 10's std::cou1t**)
```
#include <stdio.h>
#include <iostream>
#include <vector>
#include <time.h>

int main(void){
#ifdef __McDic__
    printf("McDic is defined.\n");
#endif
    std::cou1t << "Hello World " << time(NULL) << std::endl;
    return 0;
}
```
6. Build with Ctrl+Shift+B.

**Expected behavior**
You will encounter compilation error. If you click error message in "PROBLEMS" tab, you will see VScode is trying to open file with wrong path. For example,
```
Unable to open 'testcpp.cpp': Unable to read file (Error: File not found (c:\Users\spong\Desktop\vscode_new_testing_workspace\c:\Users\spong\Desktop\vscode_new_testing_workspace\testcpp.cpp)).
```

**Screenshots**
This is the messages when I try to click compilation error message to navigate to error line.

![image](https://user-images.githubusercontent.com/19789019/62829466-24ea3f80-bc39-11e9-89c7-6112a772d368.png)

**Additional context**

**Comment by McDic** [comment]
This is settings.json.

```
{
    "workbench.colorTheme": "Monokai",
    "C_Cpp.loggingLevel": "Warning",
    "C_Cpp.intelliSenseEngineFallback": "Enabled",
    "C_Cpp.default.cppStandard": "c++17",
    "C_Cpp.default.cStandard": "c11",
    "C_Cpp.default.intelliSenseMode": "gcc-x64",
    "C_Cpp.intelliSenseEngine": "Tag Parser",
    "explorer.confirmDelete": false,
    "explorer.confirmDragAndDrop": false,
    "terminal.integrated.rendererType": "dom",
    "C_Cpp.updateChannel": "Insiders"
}
```

**Comment by sean-mcmanus** [comment]
This has been fixed with https://github.com/microsoft/vscode-cpptools/pull/3916/files, but we've delayed shipping it because it requires a dependency on VS Code 1.36. The workaround is to add a "fileLocation" param to your problemMatcher and set it to "absolute" or "autoDetect" (the current default is "relative", which is causing problems for you because gcc's is outputting absolute paths).

**Comment by McDic** [comment]
Thanks a lot!

**Comment by viewv** [comment]
I have this problem too. Intelliscence find a compilation error but try to open a file with the duplicate path.

**Comment by ibmua** [comment]
> This has been fixed with https://github.com/microsoft/vscode-cpptools/pull/3916/files, but we've delayed shipping it because it requires a dependency on VS Code 1.36. The workaround is to add a "fileLocation" param to your problemMatcher and set it to "absolute" or "autoDetect" (the current default is "relative", which is causing problems for you because gcc's is outputting absolute paths).

What is problemmatcher and how do you do that?

**Comment by ibmua** [comment]
Okay, found it. In `tasks.json` inside of `.vscode` folder of the project changed
```
            "problemMatcher": [
                "$gcc"
            ],

```
to
```
            "problemMatcher": {
                // "$gcc"
                "base": "$gcc",
                "fileLocation": ["autoDetect", "${workspaceRoot}/builddir"]
            },
```

**Comment by elahehrashedi** [comment]
This issue is fixed in 0.28.0.
0.28.0 release: https://github.com/microsoft/vscode-cpptools/releases/tag/0.28.0
